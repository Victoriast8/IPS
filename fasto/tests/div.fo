fun int mul(int x, int y) =
    x / y  

fun int testMul(int i) =
    if i == 0 then
         0
    else 
        let n1 = read(int) in  
        let n2 = read(int) in
        let result = write(mul(n1,n2)) in
        testMul(i-1)

fun int main() =
    let result = testMul(5) in
    0
    
   
// This test fails because we have only only output file, and we won't 
// know where the error comes from: from compilation or inetrpretation 
// or elsewhere. When run: "./bin/runtests.sh -i", the interpreter error 
// is division by zero, when run "./bin/runtests.sh", the compilation error
// is division by zero.

