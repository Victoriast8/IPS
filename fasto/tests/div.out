3 0 -2 3 "Error: Division by zero"


